Sources/TuistCore/NodeLoaders/BundleLoader.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+enum BundleLoaderError: FatalError, Equatable {
+    case bundleNotFound(AbsolutePath)
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .bundleNotFound:
+            return .abort
+        }
+    }
+
+    /// Error description
+    var description: String {
+        switch self {
+        case let .bundleNotFound(path):
+            return "Couldn't find bundle at \(path.pathString)"
+        }
+    }
+}
+
+public protocol BundleLoading {
+    /// Reads an existing bundle and returns its in-memory representation, `ValueGraphDependency.bundle`.
+    /// - Parameter path: Path to the .bundle.
+    func load(path: AbsolutePath) throws -> GraphDependency
+}
+
+public final class BundleLoader: BundleLoading {
+    /// Initializes the loader with its attributes.
+    public init() { }
+
+    public func load(path: AbsolutePath) throws -> GraphDependency {
```

Comment:
I'd add a unit test that ensures that this method raises when the bundle doesn't exist.